ClassPaulita/GuessTheCardGame.cs

```diff
@@ -1,107 +1,174 @@
+        }
 
-    private bool IsValidInput(string input)
-    {
-        if (input == null || input.Length < 1)
-            return false;
+        private bool UserchoseToSeeInstructions(char userChoice)
+        {
+            return userChoice == 'i' || userChoice == 'I';
+        }
 
-        char userChoice = input[0];
+        private bool UserChosetoSeeBoard(char userChoice)
+        {
+            return userChoice == 'b' || userChoice == 'B';
+        }
 
-        if (UserChoseToPlay(userChoice) || UserchoseToSeeInstructions(userChoice) || UserChosetoSeeBoard(userChoice) || UserChoseToQuit(userChoice))
+        private bool UserChoseToQuit(char userChoice)
         {
-            return true;
+            return userChoice == 'q' || userChoice == 'Q';
         }
-       
-        return false;
-    }
 
-    private string PrintMenu()
-    {
-        string spacing = "\n\t\t\t\t\t\t";
+        private bool IsUserChoiceValid(string input)
+        {
+            if (input == null || input.Length < 1)
+                return false;
 
-        string menu = $"{spacing}Press P To Play" +
-                      $"{spacing}Press I For Instructions" +
-                      $"{spacing}Press B To See The High Score Board" +
-                      $"{spacing}Press Q To Quit";
+            char userChoice = input[0];
 
-        Console.WriteLine(menu);
+            if (UserChoseToPlay(userChoice) || UserchoseToSeeInstructions(userChoice) || UserChosetoSeeBoard(userChoice) || UserChoseToQuit(userChoice))
+            {
+                return true;
+            }
 
-        string userChoice = Console.ReadLine();
+            return false;
+        }
 
-        while (!IsValidInput(userChoice))
+        private string PrintMenu()
         {
-            Console.WriteLine("\n\tOoooops! You pressed an invalid option. Please try again.");
+
+            string menu = $"{spacing}Press P To Play" +
+                          $"{spacing}Press I For Instructions" +
+                          $"{spacing}Press B To See The High Score Board" +
+                          $"{spacing}Press Q To Quit";
 
             Console.WriteLine(menu);
 
-            userChoice = Console.ReadLine();
-        }
+            string userChoice = Console.ReadLine();
 
-        return userChoice;
-    }
+            while (!IsUserChoiceValid(userChoice))
+            {
+                Console.WriteLine("\n\tOoooops! You pressed an invalid option. Please try again.");
 
-    public void Run()
-    {
-        while (true)
-        {
-            string userChoice = PrintMenu();
+                Console.WriteLine(menu);
+
+                userChoice = Console.ReadLine();
+            }
+
+            return userChoice;
+        }
 
-            if (UserChoseToPlay(userChoice[0]))
+        public void Run()
+        {
+            while (true)
             {
-                Play();
+                string userChoice = PrintMenu();
```

Comment:
Also, why not make PrintMenu return a char, so that you aren't having to index it so many times and you don't have to worry about an out of index exception if PrintMenu ever gave you an empty string.